### PR TITLE
New cluster shape filter for phase2 + adding PixelPair

### PR DESCRIPTION
**Pull request by VinInn**
This Pr includes three main changes (all connected )
1) add a New cluster shape filter for phase2 
1a) I explicitly added two files for two efficiency WP 99.5 and 99.9 the default (at the moment a copy) is 99.5
2) add (back) a PixelPair iteration to recover any inefficiency  of the triplet step
(so PixelPair cover only the regions not covered by quadruplets)
3) make HighPtTriplets and PixelPair seed-extensions...

MTV
1000 single muon (blue pre5, red WP99p5, orange Wp99p9, black this PR) 
http://innocent.home.cern.ch/innocent/RelVal/Muon10csf99/plots_ootb/effandfake1.pdf
250 ttbar PU200  (blue pre5, red WP99p9, black Wp99p5, orange this PR) 
http://innocent.home.cern.ch/innocent/RelVal/ttbar200PUcsf/plots_ootb/effandfake1.pdf

timing ttbar PU200 tracking only +DQM
900_pre5
```
TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 202.108402
TimeReport      event loop Real/event = 35.805349
TimeReport     sum Streams Real/event = 203.915857
```
this PR
```
TimeReport ---------- Event  Summary ---[sec]----
TimeReport       event loop CPU/event = 146.413452
TimeReport      event loop Real/event = 25.872114
TimeReport     sum Streams Real/event = 148.02895
```

obsoletes #17810

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild please test

**Comment by VinInn** [comment]
assign upgrade

**Comment by VinInn** [comment]
@kandrosov @venturia

**Comment by VinInn** [comment]
@boudoul

**Comment by VinInn** [comment]
@ebrondol

File: RecoTracker/IterativeTracking/python/HighPtTripletStep_cff.py [hunk 1]
```diff
@@ -150,10 +158,12 @@
 )
 trackingPhase1PU70.toModify(highPtTripletStepTrajectoryBuilder,
     MeasurementTrackerName = '',
-    maxCand = 4,
```
Comment (ebrondol) [inline RecoTracker/IterativeTracking/python/HighPtTripletStep_cff.py]: @makortel do you want to change this? I thought that this era was frozen.

Comment (makortel) [inline RecoTracker/IterativeTracking/python/HighPtTripletStep_cff.py]: Doesn't matter, `trackingPhase1PU70` is not used in any workflow, and I'll clean it up sometime soon.

**Comment by ebrondol** [comment]
Thanks @VinInn . Can you please describe the legend in the files you have posted?

**Comment by slava77** [comment]
assign upgrade

**Comment by slava77** [comment]
@VinInn 
what is the purpose of the new files?
They are not used in this PR.

Is it "for upcoming developments"?

**Comment by VinInn** [comment]
I added the files to allow people to easily test different options changing just the config...

**Comment by slava77** [comment]
Some plots from PU200 (baseline 900pre6):

some large reduction of "bad" PVs
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_log10recovertexs_offlineprimaryvertices__reco_obj_zerror

Pretty large reduction in electron seeds (/2.5) with better matching ones retained
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recoelectronseeds_electronmergedseeds__reco_obj_drz1

downstream we have two times less electronGsfTracks
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recogsftracks_electrongsftracks__reco_obj_eta


ecal driven electron reduction is mainly in the barrel (apparently mostly fakes; based on MTV GSF eff plots)
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recogsfelectrons_ecaldrivengsfelectrons__reco_obj_eta

Inclusive muon distribution is still full of fakes [a part of it is a feature of what's reconstructed/saved, not necessarily fake tracks]
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recomuons_muons__reco_obj_eta

number of generalTracks is down (almost all of this reduction is from tracks with less than 7 hits)
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recotracks_generaltracks__reco_obj_eta

PF CH (~high purity tracks) are less affected suggesting relatively high purity here already
all_sign880vsorig_ttbar14tev2023d4timingpu200localwf23034p0c_recopfcandidates_particleflow__reco_obj_eta43

.... ..... ..... ......

DQM plots:

wf 21211 (10 muons extended eta range)
muon efficiencies have some visible loss in eta>3.5, while below that the efficiency looks about the same (a bit up/down in a few bins)
wf21211_hp_eff_eta

In PU200 ttbar
track fake rate is down by x2 or even more in some regions [in line with the supplied MTV plots]
mainly at low pt
wf23034pu200_hp_fake_eta
wf23034pu200_hp_fake_eta

PV fake rate is down
wf23034pu200_offpv_fake_vs_numv

b-tagging is improved moderately
wf23034pu200_csvv2_l_v_b

.... ..... ..... ......

On the technical side, based on PU200:
- miniAOD output size is down by 5%; RECO (which is actually FEVT) is down by 10%
- total CPU time in reco+miniAOD is reduced by 1.5 from 600 to 400 s/event [in my 10HS06 seconds]
    - iterative tracking "proper" (\*Step\* module names from tracking) is down by a bit more than twice: from 270 s to 120 s/evt 
    -  note that this check is before 4D PV timing went up by x3; after this PR in 91X it's probably heavier than all of tracking
- memory is down by about 200 MB/core (the job includes validation sequences; proper reco may reduce by less)

Overall, looks like a very welcome change.

**Comment by kpedro88** [comment]
Indeed, a nice improvement both for physics and resource usage

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17891	bb2e881
- implemented as described; main improvement are from the cluster shape cuts with some recovery from the pixel doublets
- jenkins tests pass and comparisons with baseline show changes only in 2023 workflows; changes start from tracking as expected
- local tests with 10-mu and TTbarPU200 samples show improvements in behavior, as posted above

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by ebrondol** [comment]
@slava77 did you also performed the check on D11? Or just a rough idea if my report in PR #17919 is compatible with your report?

**Comment by slava77** [comment]
On 3/17/17 2:58 AM, ebrondol wrote:
> @slava77 <https://github.com/slava77> did you also performed the check
> on D11? Or just a rough idea if my report in PR #17919
> <https://github.com/cms-sw/cmssw/pull/17919> is compatible with your report?


No, I did not check D11.

Remind me please the significance of it.

From README:
D4 = T3+C2+M1+I1
D11 = T5+C2+M1+I1

and
T3: Phase2 tilted tracker (v3.6.5) w/ phase 2 pixel (v4.0.2.6)
T5: Phase2 tilted tracker (v3.6.5) w/ phase 2 pixel (v4.0.2.5)
{the pixel (v4.0.2.5) has one more FPIX disk}

Is this 8+4 pixel going to become the new default or is it a side-test?

**Comment by ebrondol** [comment]
As far as I understood, the nomenclature T3 and T5 will remain the same with the idea of having a "validated tracker" (T3) and a "testing tracker" (T5). If T5 will succeed the tests and the validation we are going to do with the next RelVal production, it will become the new default (T3). 
@boudoul @kpedro88 please confirm if I am correct.

**Comment by slava77** [comment]
On 3/17/17 4:54 AM, ebrondol wrote:
> As far as I understood, the nomenclature T3 and T5 will remain the same
> with the idea of having a "validated tracker" (T3) and a "testing
> tracker" (T5). If T5 will succeed the tests and the validation we are
> going to do with the next RelVal production, it will become the new
> default (T3).
> @boudoul <https://github.com/boudoul> @kpedro88
> <https://github.com/kpedro88> please confirm if I am correct.

One more disk is probably an increment in cost,
but if that's becoming a default, OK, I will move to it once it's clear.
No intention so far to check two workflows in parallel from my side ...
at least not until we stop writing more than 100MB/event in step3 workflows.

**Comment by kpedro88** [comment]
@ebrondol that wasn't really my understanding... if there's going to be another new tracker, I assumed it would get its own number. What you propose could be done, but since we are preparing for more TDR productions, we need to be very careful about changing the baseline geometry.

**Comment by boudoul** [comment]
Dear all , we are keeping T3 , mostly because all production which has been done so far (and presumably more to come)  are all based on T3 - Physics results for the tracker TDR are and will be based on T3 - However T3 is not a version that eventually will be in the real detector, and we are devloping T5 with with respect- T5 will be used to derive tracker-tracking performance (eventually to be also added within the TDR)  , but indeed we should keep T3 (not merging or at least not in a short time scale defined by TDRs)

**Pull request merged**